# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBObjectStore.cpp
# Commit: 6078d5a9c010
# Full Hash: 6078d5a9c0107c815ca991893f9e48dd7b94588b
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-11 16:05:23
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Change definition of OpenCursorParams subtypes to allow reducing duplicated code r=ttung,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D40955
# ==============================================================================

diff -r f5aee1ae7282 -r 6078d5a9c010 dom/indexedDB/IDBObjectStore.cpp
--- a/dom/indexedDB/IDBObjectStore.cpp	Tue Sep 10 15:36:13 2019 +0000
+++ b/dom/indexedDB/IDBObjectStore.cpp	Tue Sep 10 15:36:09 2019 +0000
@@ -2273,16 +2273,16 @@
   OpenCursorParams params;
   if (aKeysOnly) {
     ObjectStoreOpenKeyCursorParams openParams;
-    openParams.objectStoreId() = objectStoreId;
-    openParams.optionalKeyRange() = std::move(optionalKeyRange);
-    openParams.direction() = direction;
+    openParams.commonParams().objectStoreId() = objectStoreId;
+    openParams.commonParams().optionalKeyRange() = std::move(optionalKeyRange);
+    openParams.commonParams().direction() = direction;
 
     params = std::move(openParams);
   } else {
     ObjectStoreOpenCursorParams openParams;
-    openParams.objectStoreId() = objectStoreId;
-    openParams.optionalKeyRange() = std::move(optionalKeyRange);
-    openParams.direction() = direction;
+    openParams.commonParams().objectStoreId() = objectStoreId;
+    openParams.commonParams().optionalKeyRange() = std::move(optionalKeyRange);
+    openParams.commonParams().direction() = direction;
 
     params = std::move(openParams);
   }